build-tools/packages/build-cli/src/commands/check/buildVersion.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Flags } from "@oclif/core";
+import { Package } from "@fluidframework/build-tools";
+import path from "node:path";
+
+import { PackageCommand } from "../../BasePackageCommand";
+
+export default class CheckBuildVersionCommand extends PackageCommand<
+	typeof CheckBuildVersionCommand
+> {
+	static description = `Checks that all packages have the same version set in package.json. The packages checked can be filtered by standard criteria. THIS COMMAND IS INTENDED FOR USE IN FLUID FRAMEWORK CI PIPELINES ONLY.`;
+
+	static flags = {
+		version: Flags.string({
+			description: "The version against which to check all the packages.",
+			exclusive: ["path"],
+		}),
+		path: Flags.directory({
+			description:
+				"Path to a directory containing a package. The version will be loaded from the package.json in this directory.",
+			exists: true,
+			exclusive: ["version"],
+		}),
+		fix: Flags.boolean({
+			description: "Fix invalid versions in the package.json file.",
+			default: false,
+		}),
+		...PackageCommand.flags,
+	};
+
+	private versionToCheck: string | undefined;
+
+	public async init(): Promise<void> {
+		await super.init();
+		if (this.flags.version === undefined) {
+			if (this.flags.path === undefined) {
+				this.error("Either version or path must be specified.");
+			}
```

Comment:
General feedback, not a blocker: I wasn't a big fan of this base method in oclif's `Command`. It's not immediately obvious that by default it will throw and prevent the rest of the method from running (good!); and making it not throw by setting the appropriate options feels like asking for unwanted behavior.

Reply:
Yeah, I am conflicted about it as well. I don't think it's obvious enough. Regardless, a separate PR as you said. Curious if you have suggestions for better patterns?

Unfortunately, there seems to be a typing problem with the `exit` function. I tried changing one usage in this PR to this:

```ts
this.errorLog("Version to check is undefined.");
this.exit(1);
```

But it didn't correctly narrow the type when used below, implying that the type system doesn't know that `exit` ends the process,